codegen/src/test/java/software/amazon/awssdk/codegen/docs/OperationDocProviderTest.java

```diff
@@ -0,0 +1,101 @@
+
+import java.util.Optional;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.codegen.internal.ExampleMetadataProvider;
+import software.amazon.awssdk.codegen.model.intermediate.Metadata;
+
+public class OperationDocProviderTest {
+
+    private static final String TEST_EXAMPLE_META_PATH = "software/amazon/awssdk/codegen/test-example-meta.json";
+
+    @AfterEach
+    void cleanupCache() {
+        ExampleMetadataProvider.clearCache();
+    }
+
+    @Test
+    public void exampleMetadataService_createLinkToCodeExample_withValidExample_returnsCorrectLink() {
+        ExampleMetadataProvider provider = ExampleMetadataProvider.getInstance(TEST_EXAMPLE_META_PATH);
+
+        Optional<String> result = provider.createLinkToCodeExample(createTestMetadata("s3"), "GetObject");
+
+        assertThat(result).isPresent();
+        assertThat(result.get()).isEqualTo("<a href=\"https://docs.aws.amazon.com/code-library/latest/ug/s3_example_s3_GetObject_section.html\" target=\"_top\">Code Example</a>");
+    }
+
+    @Test
+    public void exampleMetadataService_createLinkToCodeExample_withNonExistentExample_returnsEmpty() {
+        ExampleMetadataProvider provider = ExampleMetadataProvider.getInstance(TEST_EXAMPLE_META_PATH);
+
+        Optional<String> result = provider.createLinkToCodeExample(createTestMetadata("s3"), "NonExistentOperation");
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    public void registryPattern_withMultiplePaths_maintainsSeparateInstances() {
+        ExampleMetadataProvider provider1 = ExampleMetadataProvider.getInstance(TEST_EXAMPLE_META_PATH);
+        ExampleMetadataProvider provider2 = ExampleMetadataProvider.getInstance("nonexistent/path.json");
+
+        assertThat(provider1).isNotSameAs(provider2);
+
+        ExampleMetadataProvider provider1Again = ExampleMetadataProvider.getInstance(TEST_EXAMPLE_META_PATH);
+        assertThat(provider1).isSameAs(provider1Again);
+    }
+
+    @Test
+    public void registryPattern_threadSafety_handlesNullPath() {
+        assertThatThrownBy(() -> ExampleMetadataProvider.getInstance(null))
+            .as("getInstance should reject null paths")
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining("exampleMetaPath cannot be null");
+    }
+
+    @Test
+    public void exampleMetadataService_createLinkToCodeExample_withMedicalImagingService_returnsCorrectLink() {
+        ExampleMetadataProvider provider = ExampleMetadataProvider.getInstance(TEST_EXAMPLE_META_PATH);
+
+        Optional<String> result = provider.createLinkToCodeExample(createTestMetadata("medicalimaging"), "GetImageSet");
+
+        assertThat(result).isPresent();
+        assertThat(result.get()).isEqualTo("<a href=\"https://docs.aws.amazon.com/code-library/latest/ug/medical-imaging_example_medical-imaging_GetImageSet_section.html\" target=\"_top\">Code Example</a>");
+    }
```

Comment:
Suggesting consolidating the test with `exampleMetadataService_createLinkToCodeExample_withValidExample_returnsCorrectLink` with parameterized tests